### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by nmearl**
- Addresses #750 by giving better error messages to the `SpectralRegion` objects.
- Addresses #749 by providing a new method `SpectralRegion.from_line_list` that allows passing in the result of the `fit_lines_*` functions to produce a `SpectralRegion` object from the line list table. Added example to docs.
- Address #742 by expanding continuum function docstrings and fixing some of the failing and/or incorrect plots generated in the documentation.
- Address issues with 0.16.0 gwcs release.
- Address issue with jwst loader test which previously used the parsing utilities to automatically find flux columns but now simply accesses them directly (marked test as xfail for now).
- Fix various issues with test infrastructure; tests should now pass consistently.

### CODE REVIEW

File: .github/workflows/label_pr.yml [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-    types:
-    - opened
-
-jobs:
-  triage:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/labeler@main
-      with:
-        repo-token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (eteq) [inline .github/workflows/label_pr.yml:13]: I'm not necessarily opposed to removing this, but is the problem just that the token isn't updated?  If so maybe we should fix that instead?  Or are you actively opposed to using the labeler at all?

File: docs/fitting.rst [hunk 2]
```diff
@@ -635,47 +632,31 @@ convenience functions to perform exactly this task.  An example is shown below.
...
-
-    plt.plot(x, y)
-    plt.plot(x, y_continuum_fitted)
-    plt.title('Continuum Fitting')
-    plt.grid(True)
-
-The normalized spectrum is simply the old spectrum devided by the
-fitted continuum, which returns a new object:
-
-.. plot::
```
Comment (eteq) [inline docs/fitting.rst:669]: Why was this removed?  I think this is here specifically to demonstrate that you can do continuum normalization.

File: docs/spectral_regions.rst [hunk 3]
```diff
@@ -209,6 +209,53 @@ An example of a multiple sub-region `~specutils.SpectralRegion`:
     >>> sub_spectra[1].spectral_axis
     <SpectralAxis [34., 35., 36., 37., 38., 39., 40.] nm>
 
+
+Line List Compatibility
```
Comment (eteq) [inline docs/spectral_regions.rst:213]: Neat!  A good change, but maybe it should be called out in the changelog as a new feature since it's otherwise buried in this PR with several other things?

File: docs/spectral_regions.rst [hunk 4]
```diff
@@ -209,6 +209,53 @@ An example of a multiple sub-region `~specutils.SpectralRegion`:
...
+-----------------------
+
+`~specutils.SpectralRegion` objects can also be created from the `~astropy.table.QTable` object returned from the line
+finding routines
```
Comment (eteq) [inline docs/spectral_regions.rst:217]:
```suggestion
finding functions:
```
(quibble)

File: specutils/fitting/continuum.py [hunk 5]
```diff
@@ -21,14 +21,18 @@ def fit_generic_continuum(spectrum, median_window=3, model=Chebyshev1D(3),
...
     spectrum : Spectrum1D
         The spectrum object overwhich the equivalent width will be calculated.
-
     model : list of `~astropy.modeling.Model`
         The list of models that contain the initial guess.
-
+    median_window : float
+        The width of the median smoothing kernel used to filter the data before
+        fitting the continuum. See `~scipy.signal.medfilt` for more
+        information.
```
Comment (eteq) [inline specutils/fitting/continuum.py:29]:
```suggestion
        fitting the continuum. See the ``kernel_size`` parameter of 
        `~scipy.signal.medfilt` for more information.
```

File: specutils/fitting/fitmodels.py [hunk 6]
```diff
@@ -128,7 +128,9 @@ def find_lines_threshold(spectrum, noise_factor=1):
 
     # Threshold based on noise estimate and factor.
     uncertainty = spectrum.uncertainty
-    inds = np.where(np.abs(spectrum.flux) > (noise_factor*uncertainty.array) *
+    uncert_val = uncertainty.array if uncertainty is not None else 0
```
Comment (eteq) [inline specutils/fitting/fitmodels.py:131]:
```suggestion
    uncert_val = 0 if uncertainty is None else uncertainty.array 
```
(very minor style quibble)

**Review by eteq — CHANGES_REQUESTED** [review]
Mostly looks good, just a few questions/comments.

I also have some thoughts about improving the docstring and parsing for `SpectralRegion`, but after working on it a bit I'm realizing it's probably best as a follow-on PR

File: .github/workflows/label_pr.yml (hunk 1 above)
Comment (nmearl) [reply .github/workflows/label_pr.yml:13]: We currently don't have any actions for the labeler right now -- no configuration file that specifies what changed to look for a what labels to apply, so I'm sure what the point of having the workflow is right now.

File: docs/fitting.rst (hunk 2 above)
Comment (nmearl) [reply docs/fitting.rst:669]: Added back -- not sure how I missed it.

File: docs/spectral_regions.rst (hunk 3 above)
Comment (nmearl) [reply docs/spectral_regions.rst:213]: Yeah, there are several that I have listed that will go into the change log, but it's been happening usually as a PR before the release.

Comment (eteq) [reply docs/spectral_regions.rst:213]: I didn't realize that's when you were populating it - we should definitely move away from that - it's dangerous to rely on memory, etc to update the changelog especially with more people potentially doing releases in the future.

But so as not to delay this PR any further I'll do this myself right now!

**Comment by eteq** [comment]
Had to resolve a conflict, while adding a changelog entry, but otherwise looks good!

**Review by eteq — APPROVED** [review]
